crypto/evp/evp_rand.c

```diff
@@ -338,20 +353,33 @@ EVP_RAND_CTX *EVP_RAND_CTX_new(EVP_RAND *rand, EVP_RAND_CTX *parent)
             || !EVP_RAND_up_ref(rand)) {
         EVPerr(0, ERR_R_MALLOC_FAILURE);
         rand->freectx(ctx->data);
+        CRYPTO_THREAD_lock_free(ctx->refcnt_lock);
         OPENSSL_free(ctx);
+        EVP_RAND_CTX_free(parent);
```

Comment:
This is a down ref? maybe comment it..

Reply:
All of our free calls are down refs???

Reply:
Well, that is not true for those structures that do not have refcount but yeah, otherwise it should be implicit and we do not have comments like that everywhere where we are freeing refcounted objects. But it should be mentioned in documentation that the EVP_RAND_CTX is refcounted.

Reply:
Added "EVP_RAND_CTX are reference counted" to the documentation.